### ISSUE TITLE
sql error on a recent mysql server

### ISSUE THREAD

**Issue by tttp** [Project60/org.project60.banking#219]
Hi,

    SELECT btxb.id AS id, ba.id AS ba_id, reference, btxb.sequence as sequence, starting_date, tx_count, ba_id, ba.data_parsed as data_parsed, sum(btx.amount) as total, btx.currency as currency FROM civicrm_bank_tx_batch btxb LEFT JOIN civicrm_bank_tx btx ON btx.tx_batch_id = btxb.id LEFT JOIN civicrm_bank_account ba ON ba.id = btx.ba_id GROUP BY id ORDER BY starting_date DESC

Trying on a more recent mysql server, got that error in civibanking/extension/CRM/Banking/Page/Payments.php

I'm

**Comment by jaapjansma** [comment]
What error do you get?

**Comment by tttp** [comment]
Expression #8 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'civicrm.btx.ba_id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by]

**Comment by jaapjansma** [comment]
You could also fix this with a mysql setting in mysql.cnf under the section mysqld: 

```
[mysqld]
sql_mode=STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
```

**Comment by tttp** [comment]
I think civibanking should work on a recent mysql server without any
special configuration. Some of our users might be on a shared hosting that
doesn't allow to change it

I'm digging into the code and try to fix

**Comment by jaapjansma** [comment]
I totally agree with you.

**Comment by systopia** [comment]
PRs welcome :)

**Comment by jensschuppe** [comment]
To keep MySQL's behavior with `ONLY_FULL_GROUP_BY` enabled, the [`ANY_VALUE()`](https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_any-value) function could be used.

The correct way would be to `GROUP BY` all values that are being `SELECT`ed or to use an aggregation function for them.

**Comment by omazany** [comment]
Hi, I can confirm this "bug".  This is actually anoying, because clean installation with current cividev or CiviCRM fails with DB Errors. 

Even the current CiviCRM Developers Guide states that programmers should take this in mind - see https://docs.civicrm.org/dev/en/latest/framework/database/#only-full-group-by

There is quick and temporary fix to use CRM_Core_DAO::disableFullGroupByMode() and CRM_Core_DAO::reenableFullGroupByMode().

I will try to fix SQL queries on the dev-219 branch of my fork and will create PR. However, I am new in using github and creating PR. I will notify Bjorn when this will be ready.

If somebody is working on this fix, let me know. But I didn't find any branch with that tag.

**Comment by bjendres** [comment]
> I will try to fix SQL queries on the dev-219 branch of my fork and will create PR. However, I am new in using github and creating PR. I will notify Bjorn when this will be ready.

Sounds great, just give me a shout when it's working.

**Comment by omazany** [comment]
> 
> 
> > I will try to fix SQL queries on the dev-219 branch of my fork and will create PR. However, I am new in using github and creating PR. I will notify Bjorn when this will be ready.
> 
> Sounds great, just give me a shout when it's working.

It should be done. Works well with my data. This PR: https://github.com/Project60/org.project60.banking/pull/266

**Comment by bjendres** [comment]
Released with ``0.7``
